sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/Clauses/FromClause.cs

```diff
@@ -22,10 +22,15 @@ internal class FromClause : BaseClause
         /// Constructor for a FROM clause.
         /// </summary>
         /// <param name="collection"> Enum Collection that can be either a ADT instance or a relationship. </param>
-        public FromClause(AdtCollection collection)
+        internal FromClause(AdtCollection collection)
         {
             Type = ClauseType.FROM;
             Collection = collection;
         }
+
+        internal FromClause(string collection)
+        {
```

Comment:
you also need to set the clauseType here ... I am not sure if the clause type is even relevant anymore since you are keeping each sections' clauses in the owning class.

Reply:
That's true, we might not need it anymore. I'll keep it for the near future but make a note that if we don't use it to get rid of it.